_email.pm

```diff
@@ -0,0 +1,236 @@
+            Email::MIME->create(
+                attributes  => ATTRIBUTES->{txt},
+                body_str    => _render($tmpl_txt, $tmpl_args),
+            ),
+            Email::MIME->create(
+                attributes  => ATTRIBUTES->{html},
+                body_str    => _render($tmpl_html, $tmpl_args),
+            ),
+        ];
+        $message = Email::MIME->create(
+            header_str  => $headers,
+            attributes  => {
+                content_type => 'multipart/alternative', # parts order important
+            },
+            parts       => $parts,
+        );
+    }
+    elsif (-e $tmpl_txt) {
+        $message = Email::MIME->create(
+            header_str  => $headers,
+            attributes  => ATTRIBUTES->{txt},
+            body_str    => _render($tmpl_txt, $tmpl_args),
+        );
+    }
+    elsif (-e $tmpl_html) {
+        $message = Email::MIME->create(
+            header_str  => $headers,
+            attributes  => ATTRIBUTES->{html},
+            body_str    => _render($tmpl_html, $tmpl_args),
+        );
+    }
+    else {
+        croak sprintf '%s or %s not found', $tmpl_txt, $tmpl_html;
+    }
+
+    if (!$mail_args->{to}) {
+        my $email = Email::Abstract->new($message);
+        $mail_args->{to} = [
+            map  { $_->address                  }
+            grep { defined                      }
+            map  { Email::Address->parse($_)    }
+            map  { $email->get_header($_)       }
+            qw( to cc bcc )
+        ];
+        $message->header_set('bcc');
+    }
+    my $res = sendmail($message, $mail_args);
+    croak 'email sending failed' if !$res;
+    return $res;
+}
+
+sub _render {
+    my ($tmpl, $args) = @_;
+    state $mt = Mojo::Template->new->tag_start('[%')->tag_end('%]')->auto_escape(1);
+    local %_ = %{ $args };
+    return $mt->render_file($tmpl);
+}
+
+1; # Magic true value required at end of module
+__END__
+
+=head1 NAME
+
+_email - simple tool for sending modern looking emails
+
+=head1 SYNOPSIS
+
+    use _email;
+
+    _email::send(TEMPLATE_FILE_NAME,
+        [
+            to        => EMAIL,
+            subject   => SUBJECT,
+        ],
+        {
+            user_name => 'Someone',
+            message   => 'Hello world!',
+        },
+        {
+            from      => FROM,
+        },
+    );
+
+    file: TEMPLATE_FILE_NAME.txt.tmpl <<
+    This is simple text email template file example.
+    Welcome [%= $_{user_name} =%]
+    [%= $_{message} =%]
+    With best regards, you mailer.
+    <<
+
+    file: TEMPLATE_FILE_NAME.html.tmpl <<
+    <html>
+    <h1>Welcome [%= $_{user_name} =%]</h1>
+    <i>[%= $_{message} =%]</i>
+    With best regards, you mailer.
+    </html>
+    <<
+
+
+=head1 DESCRIPTION
+
+    Allow sending modern looking emails using mojo template
+    tools and ability apply some common/default parameters to each
+    function call.
+
+    Realized ability looking for available template files according
```

Comment:
You probably mean "Implemented", not "Realized". And it probably better to re-phrase "looking" to "auto-detecting email format based on available templates" or something like that.